text/0178-mapping-to-otlp-anyvalue.md

```diff
@@ -0,0 +1,227 @@
+# Mapping Arbitrary Data to OTLP AnyValue
+
+This document defines how to map (convert) arbitrary data (e.g. in-memory
+objects) to OTLP's AnyValue.
+
+## Motivation
+
+The mapping is necessary to correctly implement Logging Library SDKs such that
+the converted values are unambiguous and consistent across languages and
+implementations.
+
+## Explanation
+
+[AnyValue](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L27)
+is capable of representing primitive and structured data of certain types.
+
+Implementations that have a source data in any form, such as in-memory objects
+or data coming from other formats that needs to be converted to AnyValue SHOULD
```

Comment:
With the goal being to have consistency across implementation do we need to explain when it is valid to not follow this recommendation? Or, should this be made into a requirement?

Reply:
Hard to tell what such explanation can look like. I used the "SHOULD" qualifier because I can't prove these recommendations are applicable to every possible use case.

Reply:
It seems like there is a compatibility issue with implementations not conforming to this specification, different implementations for the same operations would produce different telemetry. This seems to warrant requiring these statements instead of recommending them.

This isn't a blocker. I just want to make sure we understand the implications. It is also important to keep in mind that softening a requirement to a recommendation is backwards compatible. The other way is not.